src/main/webapp/app/instructor-exercise-dashboard/instructor-exercise-dashboard.component.ts

```diff
@@ -0,0 +1,78 @@
+import {JhiAlertService} from 'ng-jhipster';
+import {Component, OnInit} from '@angular/core';
+import {ActivatedRoute} from '@angular/router';
+import {HttpResponse, HttpErrorResponse} from '@angular/common/http';
+import {Exercise, ExerciseService} from 'app/entities/exercise';
+import {InitializationState, Participation, ParticipationService} from 'app/entities/participation';
+import {Result, ResultService} from 'app/entities/result';
+import {TutorLeaderboardData} from 'app/instructor-course-dashboard/tutor-leaderboard/tutor-leaderboard.component';
+
+@Component({
+    selector: 'jhi-instructor-exercise-dashboard',
+    templateUrl: './instructor-exercise-dashboard.component.html',
+    providers: [JhiAlertService]
+})
+export class InstructorExerciseDashboardComponent implements OnInit {
+    exercise: Exercise;
+    courseId: number;
+    numberOfAssessments: number;
+
+    dataNumbersForPieChart: number[];
+    tutorLeaderboardData: TutorLeaderboardData = {};
+
+    constructor(
+        private exerciseService: ExerciseService,
+        private route: ActivatedRoute,
+        private jhiAlertService: JhiAlertService,
+        private participationService: ParticipationService,
+        private resultService: ResultService
+    ) {}
+
+    ngOnInit(): void {
+        this.courseId = +this.route.snapshot.paramMap.get('courseId');
+        this.loadExercise(Number(this.route.snapshot.paramMap.get('exerciseId')));
```

Comment:
Please use consistent means to convert `string` to `number`.

Reply:
Done